transport/http/middleware_close_response_body.go

```diff
@@ -0,0 +1,69 @@
+package http
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/awslabs/smithy-go/middleware"
+)
+
+// AddErrorCloseResponseBodyMiddleware adds the middleware to automatically
+// close the response body of an operation request if the request response
+// failed.
+func AddErrorCloseResponseBodyMiddleware(stack *middleware.Stack) error {
+	return stack.Deserialize.Insert(&errorCloseResponseBodyMiddleware{}, "OperationDeserializer", middleware.Before)
```

Comment:
yeah ... we need to close out on that, to decide where that should go, the idea could exist in smithy somewhere but would need to make sure to avoid circular dependencies what ever pkg it went into.